File: merlin-core/src/main/java/de/micromata/merlin/excel/ExcelColumnAnalyzer.java
package de.micromata.merlin.excel;

import org.apache.poi.ss.usermodel.Cell;

/**
 * Analyses all column values for some statistics.
 */
public class ExcelColumnAnalyzer extends ExcelColumnListener {
    private int maxLength = 0;

    public void readCell(Cell cell, int rowNumber) {
        String value = PoiHelper.getValueAsString(cell);
        if (value == null) {
            return;
        }
        if (value.length() > maxLength) {
            maxLength = value.length();
        }
    }

    /**
     * @return the length of the longest string value of all cells in the column.
     */
    public int getMaxLength() {
        return maxLength;
    }
}

File: merlin-core/src/main/kotlin/de/micromata/merlin/excel/ExcelColumnAnalyzer.kt
package de.micromata.merlin.excel

import de.micromata.merlin.excel.PoiHelper.getValueAsString
import org.apache.poi.ss.usermodel.Cell

/**
 * Analyses all column values for some statistics.
 */
class ExcelColumnAnalyzer
@JvmOverloads constructor(maxLength: Int = 0) : ExcelColumnListener() {
    /**
     * @return the length of the longest string value of all cells in the column.
     */
    var maxLength = maxLength
        private set

    override fun clone(): ExcelColumnAnalyzer {
        return ExcelColumnAnalyzer(maxLength)
    }

    override fun readCell(cell: Cell?, rowNumber: Int) {
        val value = getValueAsString(cell) ?: return
        if (value.length > maxLength) {
            maxLength = value.length
        }
    }

}
